### PR TITLE
Remove unnecessary null checks in `flutter_tool`

### PR DESCRIPTION
**Pull request by goderbauer**
Part of https://github.com/flutter/flutter/issues/118837.

Dart 3 drops support for non-null safe code, so we can finally turn on the unnecessary_null_comparison lint and remove the unnecessary checks it flags.

The first commit contains fixes automatically applied by running `dart fix --apply`. The second commit contains some manual fixes on top of that.

### CODE REVIEW

**Comment by christopherfujino** [comment]
oops, forgot to finish reviewing, looking now

File: packages/flutter_tools/lib/src/commands/attach.dart [hunk 1]
```diff
@@ -287,7 +287,7 @@ known, it can be explicitly provided to attach via the command-line, e.g.
 
     if ((debugPort == null && debugUri == null) || isNetworkDevice) {
       if (device is FuchsiaDevice) {
-        final String module = stringArgDeprecated('module')!;
+        final String? module = stringArgDeprecated('module');
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/attach.dart:290]: nice!

File: packages/flutter_tools/lib/src/commands/custom_devices.dart [hunk 2]
```diff
@@ -443,13 +434,13 @@ class CustomDevicesAddCommand extends CustomDevicesCommandBase {
...
 
-        final ForwardedPort forwardedPort = await (portForwarder.tryForward(port, port) as FutureOr<ForwardedPort>);
+        final ForwardedPort? forwardedPort = await portForwarder.tryForward(port, port);
         if (forwardedPort == null) {
           _printConfigCheckingError("Couldn't forward test port $port from device.",);
           result = false;
         }
 
-        await portForwarder.unforward(forwardedPort);
+        await portForwarder.unforward(forwardedPort!);
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/custom_devices.dart:443]: Can you move this to an `else` block, lest we hit a _CastError with the bang operator?

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 3]
```diff
@@ -598,11 +598,11 @@ class UpdatePackagesCommand extends FlutterCommand {
       }
     }
 
-    for (_DependencyLink path in paths) {
+    for (_DependencyLink? path in paths) {
       final StringBuffer buf = StringBuffer();
       while (path != null) {
         buf.write(path.to);
-        path = path.from!;
+        path = path.from;
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/update_packages.dart:605]: LOL, I'm guessing if you tried to run this before this change you'd get a _CastError, good catch!

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/commands/update_packages.dart:605]: (this is a maintainer tool for diagnosing version solving problems, so not often run)

Comment (goderbauer) [reply packages/flutter_tools/lib/src/commands/update_packages.dart:605]: Hehe. I was very curious about how this piece of code was running before...

File: packages/flutter_tools/lib/src/commands/custom_devices.dart (hunk 2 above)
Comment (goderbauer) [reply packages/flutter_tools/lib/src/commands/custom_devices.dart]: Done.

**Review by christopherfujino — APPROVED** [review]
LGTM! Thanks for this huge cleanup, including some bug fixes!

**Pull request merged**